### ISSUE TITLE
Remove / Default metric hyperparameter from collaborative filtering

### ISSUE THREAD

**Issue by cdbethune** [uncharted-distil/distil-primitives#59]
The `DistilCollaborativeFiltering` primitive has no default value set for the `metric` hyper-parameter which is a str value. Internally it looks like the primitive requires that value be set to `meanAbsoluteError`:

https://github.com/uncharted-distil/distil-primitives/blob/6f126d40bc8cfd2866c9477e7a0b1778a111487c/distil/modeling/collaborative_filtering.py#L54-L59

We should probably just remove the hyperparameter since we don't do anything meaningful with it at lower levels.  If kept, it should probably be converted to the core `Enum` type so that TA2s can iterate over the values.

**Comment by BalazsHoranyi** [comment]
removed this hyperparam in branch https://github.com/uncharted-distil/distil-primitives/tree/eval_check. branches got a little entangled by accident, so going to wait to PR this until some more things get merged.
